### PR TITLE
[Tests] Avoid warning from PL.

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Context:** PennyLaneDeprecationWarning: qml.transforms.hamiltonian_expand is deprecated and will be removed in version 0.39.
Instead, use qml.transforms.split_non_commuting, which can handle the same measurement type.

**Description of the Change:** Change hamiltonian_expand for split_non_commuting

**Benefits:** No warning in tests.

**Possible Drawbacks:** We won't be testing hamiltonian_expand.

### CODE REVIEW

**Review by rmoyard — APPROVED** [review]
Thanks @erick-xanadu !

**Comment by rmoyard** [comment]
@erick-xanadu It yields wrong results with Kokkos

**Comment by lillian542** [comment]
@dime10 They should be 🤔 The failing test for Kokkos is very suprising to me.

**Comment by dime10** [comment]
> @dime10 They should be 🤔 The failing test for Kokkos is very suprising to me.

Ok, just wanted to make sure we didn't replace one function with another without being sure it fits our requirements to the same degree :)

**Comment by erick-xanadu** [comment]
My guess is that this may be due to the difference in the device API?

**Comment by lillian542** [comment]
> My guess is that this may be due to the difference in the device API?

I guess seems to be something about how the old device API, QJIT and the transform interact? I've sanity-checked what I can think of in PennyLane, and would still recommend the `split_non_commuting` transform to replace `hamiltonian_expand`. I'm not sure where the discrepancy is coming up, but it only happens with `kokkos` and `split_non_commuting`, while using `qjit`.

**Comment by erick-xanadu** [comment]
@lillian542, @mehrdad2m and I will be validating whether or not this due to the new device API today.

**Comment by mehrdad2m** [comment]
> @lillian542, @mehrdad2m and I will be validating whether or not this due to the new device API today.

Hi @lillian542, @erick-xanadu. I confirmed that the test failure will be resolved by just bumping up the pennylane, lightning, and kokkos versions. This should happen after the following PR is merged:  https://github.com/PennyLaneAI/catalyst/pull/1035

**Comment by erick-xanadu** [comment]
@mehrdad2m has confirmed that running the test with kokkos in this PR #1035 then the test passes in kokkos.

**Comment by lillian542** [comment]
@erick-xanadu @mehrdad2m - awesome, glad to hear it!

**Comment by lillian542** [comment]
This is ready to go ahead now, right? The transforms are [being removed from PennyLane](https://github.com/PennyLaneAI/pennylane/pull/6204) soon.

**Comment by erick-xanadu** [comment]
@lillian542 let's wait until the rc is merged and then we will trigger the build again :)

**Comment by erick-xanadu** [comment]
@lillian542 all good!

**Pull request merged**
